### ISSUE TITLE
TA comments 

### ISSUE THREAD

**Issue by ChadFibke** [STAT545-UBC-students/hw07-zeeva85#2]
# **Coding Style**

### *The good* :thumbsup:

* Function is annotated!

### *Needs work* :thumbsdown:

* NA

# **Coding Strategy**

### *The good* :thumbsup:

* Excellent documentation of the new gap2 and gap3 datasets (explaining what the variables are)!

* Alright use of test that for your new function.

* Interesting use of prompting the user to input countries they want to look at!

* Yay no fatal errors when I run `check()`. Yes there are warnings that are getting upset with the interactiveness, and how the documentation doesn't like how your usage deviates from your argument section.   

### *Needs work* :thumbsdown:

* Your Vignettes for the gapminderplus package are not showing up as a link in your package index. (Your package index is the overarching description file when you search for your package in the package tab and then click on your package name). I think you may have forgot to run the following code because I don’t see a inst/doc directory or a vignettes directory in your repo, which holds all your vignettes md/Rmd/html that are linked to your package documentation :

```
use_vignette("vignette_name") 
## knit the vignette_name.Rmd
build_vignettes()  
```
* Your README contains the examples for your package, but most R users do not look at the github. They usually download the package from CRAN and play with the data in R. For example did you look at ggplot2 or dplyr's github accounts before downloading? It is always good to have everything bundled so that people who work in R can have all the information they need by navigating the package description.

* You did use testthat, but there are a diverse set of expect_* functions that could more holistically test your code instead of testing expect_equal once.

* It would also be good to add some internal check in your code that can provide the user with more informative error messages . For example add something like this to your functions code:

```
if (condition) {

stop("function_name requires properly spelled countries, we could not find ____"

}

# I don't know how to access the prompt input off the top of my head, but you would want to take the prompt input and put the character string into the printing message!
```
For example I tried typing in "Canada", and a random strings of text "asdfarg" and the function still ran and did't say anything about a potential error.

* Also you may want to make the scanning case insensitive. "canada" or "Canada"

* As of now this function is cool, but it prevents automation. What if I wanted 15 different countries? I would rather have a function that grabs the countries of interest and feeds them into the prompt for me.

# **Graphs & Tables**

### *The good* :thumbsup:

* NA

### *Needs work* :thumbsdown:

* NA

# **Creativity & Ease of Access**

### *The good* :thumbsup:

* Easy navigation from canvas to your repo.

* The README is populated with informative details about installation and function use. Also shows the new functions in action!  

* Got creative with the assignment and make an extension to gapminder! :fire:

* Function in interactive and fun!

### *Needs work* :thumbsdown:

* The read me should explain the contents of the package, and how to install (which you do) *but* you should also include a selling point! Why is your package better than using base r, or just gapminder?

**Comment by zeeva85** [comment]
@ChadFibke, Thank you for those informative comments, although the assignment is over I actually intend to fix those warnings and notes, could you kindly let me know what the warning means and how i fix it, I tried for hours, I got close but i didn't quite understand how to fix the warning and notes.

>Yay no fatal errors when I run check(). Yes there are warnings that are getting upset with the interactiveness, and how the documentation doesn't like how your usage deviates from your argument section.

 I will find you in class so you could explain what the warning and the notes.

Thank you !!!

**Comment by ChadFibke** [comment]
Hey @zeeva85, I will be attending Thursday's class (Nov 22th). I will see you then!

**Comment by zeeva85** [comment]
@ChadFibke 

I have always found your feedbacks and marking consistent, could I have a second opinion on my homework 8, I am having though time understanding the evaluation as I did not get feedback.

TQ

**Comment by ChadFibke** [comment]
Hey @zeeva85 ,

Let me take a look

**Comment by zeeva85** [comment]
Thank you @ChadFibke, much appreciated.

**Comment by ChadFibke** [comment]
@zeeva85 ,

The comments are not very concise. I will Email your TA to ask for more informative feedback. I've noticed some back and forth between you guys and that seems to be going nowhere. If the TA does not get back to you by Wednesday let me know and I'll give you more feedback. 

Chad

**Comment by zeeva85** [comment]
Yes @ChadFibke , I do not understand how/why marks were deducted in every sections available, even for sections such as creativity or accessibility. 

Could you kindly have a look at the rubrics and marks allocated to me and state If it makes sense to you and kindly help me understand what did I miss in those sections to loose marks.

Thank you and I will revert back

**Comment by ChadFibke** [comment]
@zeeva85,

I have reached out to your TA to see if they can provide more informative feedback. Your assignment has minor errors that I can see why the TA has docked you marks (in each section). However, I do not want to step on the other TA's toes. Give the TA some time to respond. Again if the TA doesn't get back to you by Wednesday I will fill in the blanks!

Chad

**Comment by zeeva85** [comment]
@ChadFibke 

Yes, I agree with some of them, but some are bizarre considering the` include_CSS` wasn't considered in the grading according to TA.

It is a common practice to justify why marks is deducted in every section. If no feedback is provided in for every section of why the marks is deducted then it shouldn't be deducted cause there is no justification for it. Furthermore the justification is also to avoid marks being deducted twice in two different sections for the same issue which is inappropriate, at least that was what I understood upon completing the BioTAp course at UBC.

I understand, will wait for them and will revert on Wed if nothing shows.

Thank you
